test/fiddle/test_pointer.rb

```diff
@@ -170,6 +186,31 @@ def test_free=
       assert_equal free.ptr, ptr.free.ptr
     end
 
+    def test_free!
+      ptr = Pointer.malloc(4)
+      refute ptr.freed?
+      ptr.free!
+      assert ptr.freed?
+      ptr.free!                 # you can safely run it again
+      assert ptr.freed?
+
+      ptr = Pointer.malloc(4, Fiddle::RUBY_FREE)
+      refute ptr.freed?
+      ptr.free!
+      assert ptr.freed?
+      ptr.free!                 # you can safely run it again
+      assert ptr.freed?
+      GC.start                  # you can safely run the GC routine
+      assert ptr.freed?
```

Comment:
Could you split this case to another test?

Reply:
Done.